src/types.rs

```diff
@@ -1076,6 +1085,207 @@ pub(crate) fn adjust_mappings(
     new_tokens
 }
 
+pub fn adjust_mappings_from_multiple(
+    mut this: SourceMap,
+    mut input_maps: Vec<crate::lazy::SourceMap>,
+) -> SourceMap {
+    // Helper struct that makes it easier to compare tokens by the start and end
+    // of the range they cover.
+    #[derive(Debug, Clone, Copy)]
+    struct Range<'a> {
+        start: (u32, u32),
+        end: (u32, u32),
+        value: &'a RawToken,
+        map_idx: u32,
+    }
+
+    /// Turns a list of tokens into a list of ranges, using the provided `key` function to determine the order of the tokens.
+    #[allow(clippy::ptr_arg)]
+    fn create_ranges(
```

Comment:
Should we reuse this function?

Reply:
The new version has a `map_idx: u32` that needs to be threaded through, so they are not identical.